services/github/github-package-json.service.js

```diff
@@ -0,0 +1,156 @@
+'use strict'
+
+const Joi = require('joi')
+const { renderVersionBadge } = require('../../lib/version')
+const {
+  transformAndValidate,
+  renderDynamicBadge,
+} = require('../dynamic-common')
+const { semver } = require('../validators')
+const { ConditionalGithubAuthService } = require('./github-auth-service')
+const { fetchJsonFromRepo } = require('./github-common-fetch')
+const { documentation } = require('./github-helpers')
+
+const versionSchema = Joi.object({
+  version: semver,
+}).required()
+
+class GithubPackageJsonVersion extends ConditionalGithubAuthService {
```

Comment:
I broke this out into its own file in anticipation of adding a bunch of package.json-specific features, which move it away from the manifest badge. Most of the logic has been moved out into functions, though there's a bit of duplication of wiring.